### PR TITLE
Fix Asset Store flaky test

### PR DESCRIPTION
**Pull request by m00g3n**
Fix astore flaky test

Changes proposed in this pull request:
- Add retries with exponential backoff in asset-store tests 
- Optimize communication with api-server

See also: #5913

### CODE REVIEW

File: resources/assetstore/templates/tests/rbac.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ rules:
     nonResourceURLs: []
   - apiGroups: ["assetstore.kyma-project.io"]
     resources: ["buckets", "clusterbuckets", "assets", "clusterassets"]
-    verbs: ["create", "get", "delete"]
+    verbs: ["create", "get", "delete", "watch", "list", deletecollection]
```
Comment (aerfio) [inline resources/assetstore/templates/tests/rbac.yaml]: image
looks like a typo

Comment (m00g3n) [reply resources/assetstore/templates/tests/rbac.yaml]: it's one of the `API request verbs`:
https://kubernetes.io/docs/reference/access-authn-authz/authorization/#review-your-request-attributes;

File: tests/asset-store/pkg/resource/resource.go [hunk 2]
```diff
@@ -1,73 +1,131 @@
...
+		return nil
+	}, callbacks...)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			r.log("Cannot delete. Resource %s with name '%s' is not found.", r.kind, name)
-			return nil
+		return err
```
Comment (michal-hudy) [inline tests/asset-store/pkg/resource/resource.go]: Can you wrap errors to have a context?

File: tests/asset-store/pkg/namespace/namespace.go [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
 
-func (n *Namespace) Create() error {
-	_, err := n.coreCli.Namespaces().Create(&v1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:  n.name,
-		},
-	})
-
+func (n *Namespace) Create(callbacks ...func(...interface{})) error {
+	err := retry.OnIsAlreadyExist(retry.DefaultBackoff, func() error {
```
[the remark this answers is not in the document]
Comment (m00g3n) [reply tests/asset-store/pkg/namespace/namespace.go]: I've already renamed it to 
```
func WithIgnoreOnIsAlreadyExist(backoff wait.Backoff, fn func() error, callbacks ...func(...interface{})) error {
	return retry.OnError(backoff, errorFn(callbacks...), fnWithIgnore(fn, errors.IsAlreadyExists, callbacks...))
}
```

**Pull request merged**